packages/components/pages/f-mfa/src/components/Mfa.vue

```diff
@@ -207,18 +243,60 @@ export default {
             console.log('DEBUG - onShowHelpInfo'); // eslint-disable-line no-console
         },
 
-        // To be replaced by API call
-        postValidateMfaToken () {
-            const err = new Error('TEST - 400 error');
-            err.response = {
-                status: 400
-            };
-            throw err;
+        /**
+        * If no error has already occurred it reads, decodes, trims, validates the query
+        * string parameters based on the regex provided. Then store the value against the
+        * supplied property indicated by the key.
+        * If the key is not present in the query string or the value is not valid, the
+        * showErrorPage flag is raised and the issue is logged.
+        *
+        * @param {string} key - The key to search for in the query string.
+        * @param {string} field - The data field to store the value against.
+        * @param {string} regex - The regex to validate the value against.
+        */
+        processQueryString (key, field, regex) {
+            if (this.showErrorPage === true) {
+                return;
+            }
+            let err = new Error(`Missing key in querystring or failed regex [${key}]`);
```

Comment:
Should the creation of this be moved to within the error block? Also how does this work as it looks like it is reassigned to `error` later?

Reply:
Is does need to be declared outside but now I look at it I can give a better error message if I assign inside 👍